### PR TITLE
#164829320 - Configure Signup

### PR DESCRIPTION
**Pull request by kihuha**
#### What does this PR do?
Configures the user signup page

#### Description of the task to be completed
- Render a signup form
- Validate user input 
- Render errors returned from the API

#### How should this be manually tested?
- Clone the repo and navigate to the root directory using a terminal
- Install all the dependencies by typing `npm install`
- Start the server by typing `npm start`
- Navigate to http://localhost:3000/signup using your browser
- To run the test suite by typing `npm test`
- To view the coverage, type `npm test -- --coverage`

#### Relevant PT stories
https://www.pivotaltracker.com/story/show/164829320

### CODE REVIEW

File: src/containers/signup/Signup.js [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+    );
+
+    return res
+  }
+
+  handleSubmit(e) {
+    e.preventDefault();
+
+    if (this.state.password !== this.state.confirmPassword) {
+      this.setPasswordState(['Passwords do not match'], true);
```
Comment (kirega) [inline src/containers/signup/Signup.js]: @kihuha This section is buggy, the if statement does not exit the program flow on the event of an unmatching password and confirm password. As such while testing this functionality I discovered that the form submitted despite the passwords not matching. 
As a solution, I discovered that you can solve by using `return   this.setPasswordState(['Passwords do not match'], true);`

Comment (kihuha) [reply src/containers/signup/Signup.js]: Thanks for the feedback. I have added a return statement to exit the 'if' statement if passwords do not match

**Review by MainaKamau92 — DISMISSED** [review]

**Review by Raywire — DISMISSED** [review]
Functionality is on point.

**Review by MainaKamau92 — DISMISSED** [review]

**Review by MainaKamau92 — APPROVED** [review]

**Review by Raywire — APPROVED** [review]
Nice work on the feature.

**Pull request merged**